pkg/amazon/backend/up.go

```diff
@@ -96,7 +99,29 @@ func (b *Backend) Up(ctx context.Context, options *cli.ProjectOptions) error {
 		b.Down(ctx, options)
 	}()
 
-	return b.WaitStackCompletion(ctx, project.Name, operation, w)
+	err = b.WaitStackCompletion(ctx, project.Name, operation)
+	// update status for external resources (LB and cluster)
+	loadBalancerName := fmt.Sprintf("%.32s", fmt.Sprintf("%sLoadBalancer", strings.Title(project.Name)))
+	for k := range template.Resources {
+		switch k {
+		case "Cluster":
+			if cluster == "" {
+				continue
+			}
+		case loadBalancerName:
+			if lb == "" {
+				continue
+			}
+		default:
+			continue
+		}
+		b.writer.Event(progress.Event{
+			ID:         k,
+			Status:     progress.Done,
+			StatusText: "",
+		})
+	}
```

Comment:
I don't understand this part. If cluster or LB are set by user ("external"), those won't appear in the template as `Resources`. Also, better use resource.AWSType() on such a switch statement.

Reply:
We don't get any event related to it to update it and stays in 'pending' until the end. And the progress would show 12/13 out of all done. It is just something to give us consistency in the progress output, cosmetic thing.

Reply:
But why does it appear on the progress list? This is a provided resource, it has no matching entry in the template Resources map, right?

Reply:
It is added in the template resource map under the `Cluster` key and it always appeared to me in the progress output even before this change, and it looked strange as the state of it was never updated.

Reply:
ah ok, that's because this one is controlled in the template by a `Condition` (if parameter is set, resource is not ran).

Reply:
please note same issue applies to securityGroups

Reply:
a possible workaround is to not set the initial content of the status as `Pending` and just wait for cloudformation event. Main reason I did this was for flacky ANSI code issues to get a correct rendering without having time to investigate. ProgressWriter seems more robust :p